bin/fin

```diff
@@ -6287,6 +6344,631 @@ set_project_root_permissions ()
+}
+
+# Runs a generic drush rsync command.
+provider_pull_generic_drush_rsync ()
+{
+	echo-yellow "Downloading files from ${HOSTING_SITE} on the ${HOSTING_ENV} environment"
+	[[ ! -d ${PROJECT_ROOT}/${DOCROOT}/${FILES_DIR} ]] && mkdir -p ${PROJECT_ROOT}/${DOCROOT}/${FILES_DIR} > /dev/null
+	_exec "drush -y rsync \
+		@${HOSTING_SITE}.${HOSTING_ENV}:%files \
+		${PROJECT_ROOT}/${DOCROOT}/${FILES_DIR} \
+		--delete \
+		${rsync_options}"
+	if_failed_error "Error Downloading Files from ${HOSTING_SITE} on the ${HOSTING_ENV} environment"
+}
+
+#############################
+### ACQUIA PROVIDER FUNCTIONS
+#############################
+
+# Check to see if acquia keys have been set.
+provider_pull_acquia_precheck ()
+{
+	if [[ -z "${SECRET_ACAPI_EMAIL}" ]] || [[ -z "${SECRET_ACAPI_KEY}" ]]; then
+		echo-error "SECRET_ACAPI_EMAIL and SECRET_ACAPI_KEY must be set in order to work with Acquia."
+		exit 1;
+	fi
+
+	if [[ "${ARGV[0]}" == "init" ]]; then
+		return;
+	fi
+
+	provider_pull_precheck
+}
+
+# Init process to pull code from acquia.
+provider_pull_acquia_init ()
+{
+	# Set HOSTING_ENV to dev if none are set.
+	HOSTING_ENV=${HOSTING_ENV:-dev}
+	# Get the repo url for the specific environment.
+	local repo_url=$(run_cli -e COLUMNS="200" "drush -q acquia-update; drush @${HOSTING_SITE}.${HOSTING_ENV} ac-site-info | grep 'vcs_url' | awk '{print \$3}' | tr -d '[:space:]'")
+	if_failed_error "Error accessing Acquia Cloud API to get Project Git Repo"
+	# Get the branch for the specific environment.
+	local repo_checkout=$(run_cli -e COLUMNS="200" "drush -q acquia-update; drush @${HOSTING_SITE}.${HOSTING_ENV} ac-environment-info | grep 'vcs_path' | awk '{print \$3}' | sed 's/tags\///g' | tr -d '[:space:]'")
+	if_failed_error "Error accessing Acquia Cloud API to get ${HOSTING_ENV} info."
+
+	echo-yellow "Cloning Acquia Site Locally: ${HOSTING_SITE} - ${HOSTING_ENV} in ${project_directory}"
+	# Run a git clone to get the specific repo.
+	run_cli "git clone -q --depth=1 --branch='${repo_checkout}' ${repo_url} ${project_directory}"
+	if_failed_error "Error cloning Acquia Project."
+}
+
+# Pull process to pull files from Acquia.
+provider_pull_acquia_files ()
+{
+	# Execute a generic rsync on files from remote.
+	provider_pull_generic_drush_rsync
+}
+
+# Pull process to pull db from Acquia.
+provider_pull_acquia_db ()
+{
+	# Update Acquia Drush Aliases in container.
+	_exec drush acquia-update
+
+	# TODO: Allow for all databases to be downloaded. drush ac-database-list will list all databases for a site.
+	# Following will grab all databases for a site.
+	# local acquia_dbs=$(_exec drush "@${HOSTING_SITE}.${HOSTING_ENV}" ac-database-list | awk '{print $3}')
+
+	local acquia_db_name="${ARGV[1]}"
+	if [[ -z "${acquia_db_name}" ]]; then
+		echo-error "Database name is required."
+		exit 1
+	fi
+
+	local query=$(_exec drush @${HOSTING_SITE}.${HOSTING_ENV} ac-database-instance-backup-list ${acquia_db_name});
```

Comment:
it should be split into 2 lines. Bash has a quirk that this line will always return result `0` even if command fails! That is because the last command will be `local`.

You can reproduce it yourself

```bash
#!/bin/bash

testie () {
    local abc=$(exit 1)
    echo $?
}

testie
```
So the line needs to be changed to:

```bash
local query
query=$(...)
```

You only need to do the separation if you need the correct error code.